src/core/Solver_intf.ml

```diff
@@ -271,41 +270,41 @@ module type PROOF = sig
 
   (** {3 Type declarations} *)
 
-  exception Resolution_error of string
   (** Raised when resolution failed. *)
+  exception Resolution_error of string
 
   type formula
   type atom
   type lemma
-  type clause
+
   (** Abstract types for atoms, clauses and theory-specific lemmas *)
+  type clause
 
-  type t
   (** Lazy type for proof trees. Proofs are persistent objects, and can be
       extended to proof nodes using functions defined later. *)
+  type t
 
+  (** A proof can be expanded into a proof node, which show the first step of the proof. *)
   and proof_node = {
     conclusion : clause;  (** The conclusion of the proof *)
-    step : step;          (** The reasoning step used to prove the conclusion *)
+    step : step  (** The reasoning step used to prove the conclusion *)
   }
-  (** A proof can be expanded into a proof node, which show the first step of the proof. *)
 
   (** The type of reasoning steps allowed in a proof. *)
   and step =
-    | Hypothesis of lemma
-    (** The conclusion is a user-provided hypothesis *)
-    | Assumption
-    (** The conclusion has been locally assumed by the user *)
+    | Hypothesis of lemma  (** The conclusion is a user-provided hypothesis *)
```

Comment:
I prefer it when all doc comments are in the same position (either all on the same line as the variant, or all on a separate line). That's not a hard requirement, but just a personal taste for regularity of code when reading, :/